pex/third_party/__init__.py

```diff
@@ -100,9 +100,12 @@ def containing(cls, root):
     prefix = ''
     path = root
     while path:
+      # We replace os.sep with '/' here because the zip file format spec specifies that paths must
+      # use forward slashes. See section 4.4.17 of
+      # https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT.
       if zipfile.is_zipfile(path):
-        return cls(zipfile_path=path, prefix=prefix + os.sep if prefix else '')
-      prefix = os.path.join(prefix, os.path.basename(path))
+        return cls(zipfile_path=path, prefix='{}{}'.format(prefix, os.sep).replace(os.sep, '/'))
```

Comment:
The `prefix` starts as `''` and is built up from there on the line below. Since that line takes care of `/` it seems like there is no need to `replace` here. Also, you've now changed handling of the `''` case. That case used to trigger `cls(..., prefix='')` and now you `cls(..., prefix='/')`. I think you want:
```python
return cls(zipfile_path=path, prefix='{}/'.format(prefix) if prefix else '')
```

Reply:
Good catch. Fixed!